hyperspy/drawing/_widgets/polygon.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2024 The HyperSpy developers
+#
+# This file is part of HyperSpy.
+#
+# HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with HyperSpy. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+import matplotlib
+from matplotlib.widgets import PolygonSelector
+
+from hyperspy.drawing.widget import WidgetBase
+
+from packaging.version import Version
+
+
+class PolygonWidget(WidgetBase):
+    """PolygonWidget is a widget for drawing an arbitrary
+    polygon, which can then be used as a region-of-interest.
+    The polygon can be moved by pressing 'shift' and clicking.
+    A polygon vertex can be moved by clicking its handle. If incomplete,
+    it is also necessary to press 'ctrl'.
+    The polygon can be deleted by pressing 'esc'.
+    """
+
+    def __init__(self, axes_manager, **kwargs):
+        """
+        Parameters
+        ----------
+        axes_manager : hyperspy.axes.AxesManager
+            The axes over which the `PolygonWidget` will interact.
+        """
+
+        super().__init__(axes_manager, **kwargs)
+
+        if self.axes_manager is not None:
+            if self.axes_manager.navigation_dimension > 1:
+                self.axes = self.axes_manager.navigation_axes[0:2]
```

Comment:
Please add some tests that is not currently covered by the test suite, use the codecov yellow annotation on github to find out what is not covered.

Reply:
I have added a lot more tests now. Some edge cases are still not covered, however, but the main code is covered more thoroughly.